pymeasure/instruments/hp/hp4142B.py

```diff
@@ -0,0 +1,2012 @@
+            #'HP41423A': 'HVU',
+            #'HP41424A': 'VSVMU',
+            #'HP41425A': 'AFU',
+        }
+        out = {}
+        for i, module in enumerate(modules):
+            module = module.split(',')
+            if not module[0] == '0':
+                try:
+                    out[i+1] = module_names[module[0]]
+                    # i+1: channels start at 1 not at 0
+                except Exception:
+                    raise NotImplementedError(
+                        'Module {} is not implented yet!'.format(module[0]))
+        return out
+
+    def initialize_smu(self, channel, smu_type, name):
+        """ Initializes SMU instance by calling :class:`.SMU`.
+        :param channel: SMU channel
+        :type channel: int
+        :param smu_type: SMU type, e.g. ``'HRSMU'``
+        :type smu_type: str
+        :param name: SMU name for pymeasure (data output etc.)
+        :type name: str
+        :return: SMU instance
+        :rtype: :class:`.SMU`
+        """
+        if channel in (
+                list(range(101, 1101, 100))
+                + list(range(102, 1102, 100))):
+            channel = int(str(channel)[0:-2])
+            # subchannels not relevant for SMU/CMU
+        channel = strict_discrete_set(channel, range(1, 11))
+        self._smu_names[channel] = name
+        smu_reference = SMU(self, channel, smu_type, name)
+        self._smu_references[channel] = smu_reference
+        return smu_reference
+
+    def initialize_all_smus(self):
+        """ Initialize all SMUs by querying available modules and creating
+        a SMU class instance for each.
+        SMUs are accessible via attributes ``.smu1`` etc.
+        """
+        modules = self.query_modules()
+        i = 1
+        for channel, smu_type in modules.items():
+            if 'MU' in smu_type:
+                setattr(self, 'smu'+str(i),
+                        self.initialize_smu(
+                            channel, smu_type, 'SMU'+str(i)))
+                i += 1
+
+    def pause(self, pause_seconds):
+        """ Pauses Command Excecution for given time in seconds (``PA``)
+        :param pause_seconds: Seconds to pause
+        :type pause_seconds: int
+        """
+        self.write("PA %d" % pause_seconds)
+
+    def abort(self):
+        """ Aborts the present operation but channels may still output
+        current/voltage (``AB``)
+        """
+        self.write("AB")
+
+    def force_gnd(self):
+        """ Force 0V on all channels immediately. Current Settings can
+        be restored with RZ. (``DZ``)
+        """
+        self.write("DZ")
+
+    def check_errors(self):
+        """ Check for errors (``ERR?``)
+        """
+        error = self.ask("ERR?")
+        error = error.split(",")
+        if int(error[0]) == 0:
+            return
+        else:
+            raise IOError(
+                "HP4142B Errors {0}, {1}, {2}, {3}".format(error[0], error[1], error[2], error[3]))
+
+    def check_idle(self):
```

Comment:
nowadays that exists as `Instrument.complete`, so should be removed in both classes.